src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -99,4 +101,108 @@ public static IResourceBuilder<AzureEventHubsResource> AddEventHub(this IResourc
         builder.Resource.Hubs.Add((name, configureHub));
         return builder;
     }
+
+    /// <summary>
+    /// Configures an Azure Event Hubs resource to be emulated. This resource requires an <see cref="AzureEventHubsResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure Event Hubs resource builder.</param>
+    /// <param name="storageResource">An optional Azure Storage resource.</param>
+    /// <param name="configureContainer">Callback that exposes underlying container used for emulation to allow for customization.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResourceBuilder<AzureEventHubsResource> builder, IResourceBuilder<IResource>? storageResource = null, Action<IResourceBuilder<AzureEventHubsEmulatorResource>>? configureContainer = null)
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return builder;
+        }
+
+        builder.ApplicationBuilder.Services.TryAddLifecycleHook<AzureEventHubsEmulatorConfigWriterHook>();
+
+        // Add emulator container
+
+        builder
+            .WithEndpoint(name: "emulator", targetPort: 5672)
+            .WithAnnotation(new ContainerImageAnnotation
+            {
+                Registry = "messagingemulators.azurecr.io",
+                Image = "microsoft/azure/eventhubs/emulator",
+                Tag = "latest"
+            })
+            .WithAnnotation(new ContainerMountAnnotation(
+                Path.GetFullPath(Path.GetTempFileName(), builder.ApplicationBuilder.AppHostDirectory),
```

Comment:
This already returns a full path
```suggestion
                Path.GetTempFileName(),
```

Reply:
Done